api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirEncounterServiceImpl.java

```diff
@@ -143,22 +143,28 @@ public void delete(@Nonnull String uuid) {
 	@Transactional(readOnly = true)
-	public IBundleProvider searchForEncounters(DateRangeParam date, ReferenceAndListParam location,
-	        ReferenceAndListParam participant, ReferenceAndListParam subject, TokenAndListParam encounterType,
-	        TokenAndListParam id, DateRangeParam lastUpdated, SortSpec sort, HashSet<Include> includes,
-	        HashSet<Include> revIncludes) {
-		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.DATE_RANGE_SEARCH_HANDLER, date)
-		        .addParameter(FhirConstants.LOCATION_REFERENCE_SEARCH_HANDLER, location)
-		        .addParameter(FhirConstants.PARTICIPANT_REFERENCE_SEARCH_HANDLER, participant)
-		        .addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, subject)
-		        .addParameter(FhirConstants.ENCOUNTER_TYPE_REFERENCE_SEARCH_HANDLER, encounterType)
-		        .addParameter(FhirConstants.COMMON_SEARCH_HANDLER, FhirConstants.ID_PROPERTY, id)
-		        .addParameter(FhirConstants.COMMON_SEARCH_HANDLER, FhirConstants.LAST_UPDATED_PROPERTY, lastUpdated)
-		        .addParameter(FhirConstants.INCLUDE_SEARCH_HANDLER, includes)
-		        .addParameter(FhirConstants.REVERSE_INCLUDE_SEARCH_HANDLER, revIncludes).setSortSpec(sort);
-		
-		IBundleProvider visitBundle = visitService.searchForVisits(theParams);
-		IBundleProvider encounterBundle = searchQuery.getQueryResults(theParams, dao, translator, searchQueryInclude);
+	public IBundleProvider searchForEncounters(EncounterSearchParams esp) {
+		SearchParameterMap theParams = new SearchParameterMap()
+		        .addParameter(FhirConstants.DATE_RANGE_SEARCH_HANDLER, esp.getDate())
+		        .addParameter(FhirConstants.LOCATION_REFERENCE_SEARCH_HANDLER, esp.getLocation())
+		        .addParameter(FhirConstants.PARTICIPANT_REFERENCE_SEARCH_HANDLER, esp.getParticipant())
+		        .addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, esp.getSubject())
+		        .addParameter(FhirConstants.ENCOUNTER_TYPE_REFERENCE_SEARCH_HANDLER, esp.getEncounterType())
+		        .addParameter(FhirConstants.COMMON_SEARCH_HANDLER, FhirConstants.ID_PROPERTY, esp.getId())
+		        .addParameter(FhirConstants.COMMON_SEARCH_HANDLER, FhirConstants.LAST_UPDATED_PROPERTY, esp.getLastUpdated())
+		        .addParameter(FhirConstants.INCLUDE_SEARCH_HANDLER, esp.getIncludes())
+		        .addParameter(FhirConstants.REVERSE_INCLUDE_SEARCH_HANDLER, esp.getRevIncludes())
+		        .addParameter(FhirConstants.HAS_SEARCH_HANDLER, esp.getHasAndListParam()).setSortSpec(esp.getSort());
+		
+		IBundleProvider visitBundle = new SimpleBundleProvider();
+		IBundleProvider encounterBundle = new SimpleBundleProvider();
+		
+		if (shouldIncludeResourceBasedOnEncounterTag(esp.getTag(), "visit")) {
+			visitBundle = visitService.searchForVisits(theParams);
+		}
+		if (shouldIncludeResourceBasedOnEncounterTag(esp.getTag(), "encounter")) {
+			encounterBundle = searchQuery.getQueryResults(theParams, dao, translator, searchQueryInclude);
+		}
```

Comment:
The above lines is the code responsible for handling the _tag parameter on Encounter.   Basically, if no tags restrict the encounter, or if tags that restrict match appropriately, include matching resources.  Otherwise, use an empty bundle provider.